nfs/src/rpc_task.cpp

```diff
@@ -1149,7 +1149,29 @@ void rpc_task::run_read()
              * after it acquires the lock.
              */
             found_in_cache = false;
-            read_from_server(bc_vec[i]);
+
+            /*
+             * TODO: If we have just 1 bytes chunk to fill, which is the most common case,
+             *       avoid creating child task and process evrything in this same task.
+             */
+
+            /*
+             * Create a child rpc task to issue the read RPC to the backend.
+             */
+            struct rpc_task *child_tsk =
+                get_client()->get_rpc_task_helper()->alloc_rpc_task(FUSE_READ);
+
+            child_tsk->init_read(
+                rpc_api->req,
+                rpc_api->read_task.get_ino(),
+                bc_vec[i].length,
```

Comment:
This can be parent length and offset?

Reply:
it should correspond to the individual bytes_chunk it's reading and not the exact parent read's offset and size.
Your testing would have caught this, but since you didn't actually have the case of multiple bytes_chunk, it didn't matter.